src/query.rs

```diff
@@ -89,14 +103,18 @@ unsafe impl<'a, T: Component> Fetch<'a> for FetchRead<T> {
         }
     }
 
-    fn borrow(archetype: &Archetype) {
-        archetype.borrow::<T>();
+    fn borrow(archetype: &Archetype, state: Self::State) {
+        archetype.borrow::<T>(state);
     }
-    fn new(archetype: &'a Archetype) -> Option<Self> {
-        archetype.get_base::<T>().map(Self)
+    #[allow(clippy::needless_question_mark)]
+    fn prepare(archetype: &Archetype) -> Option<Self::State> {
+        Some(archetype.get_state::<T>()?)
```

Comment:
Why violate `clippy::needless_question_mark`?

Reply:
Ah, this was a false-positive in an intermediate state (because there was the decay of a reference into a raw pointer between that question mark and `Some`), but since `Self::State` is now `usize` this should go away.